### PR TITLE
[TEST] Add hierarchical_binning test with a max split bin at position 0.

### PR DESCRIPTION
**Pull request by smehringer**

### CODE REVIEW

File: test/api/pack/hierarchical_binning_test.cpp [hunk 1]
```diff
@@ -79,6 +79,35 @@ TEST(hierarchical_binning_test, another_example)
...
+    std::stringstream header_buffer;
+    pack_data data;
+    data.output_buffer = &output_buffer;
+    data.header_buffer = &header_buffer;
+    data.filenames = {"seq0", "seq1", "seq2", "seq3"};
+    data.kmer_counts = {500, 500, 500, 500};
+    data.compute_fp_correction(0.05, 2);
+
+    hierarchical_binning algo{data, config};
+    EXPECT_EQ(std::get<0>(algo.execute()), 0); // #HIGH_LEVEL_IBF max_bin_id:1
```
Comment (eseiler) [inline test/api/pack/hierarchical_binning_test.cpp:97]:
```suggestion
    EXPECT_EQ(std::get<0>(algo.execute()), 0); // #HIGH_LEVEL_IBF max_bin_id:0
```

File: test/api/pack/hierarchical_binning_test.cpp [hunk 2]
```diff
@@ -79,6 +79,35 @@ TEST(hierarchical_binning_test, another_example)
     EXPECT_EQ(header_buffer.str() + output_buffer.str(), expected_file);
 }
 
+TEST(hierarchical_binning_test, high_level_max_bin_id_is_0)
```
Comment (eseiler) [inline test/api/pack/hierarchical_binning_test.cpp:82]: Shouldn't this one be called `only_top_level` or something? Or is it important that the `max_bin_id` is exactly `0`

**Review by eseiler — CHANGES_REQUESTED** [review]
the commit/pr title seems a bit wrong? There is no split bin?

File: test/api/pack/hierarchical_binning_test.cpp (hunk 2 above)
Comment (smehringer) [reply test/api/pack/hierarchical_binning_test.cpp:82]: Yes. It is the edge case that the trace it at position 0, the very first bin, and that is a split bin (rather not-merged-bin because it is a single bin, I used to call them split bins anyway) and that has the highest weight.

Comment (eseiler) [reply test/api/pack/hierarchical_binning_test.cpp:82]: It's ok, but add what you just wrote as comment to the test?

Comment (smehringer) [reply test/api/pack/hierarchical_binning_test.cpp:82]: sure
